dbus_service.py

```diff
@@ -598,11 +604,40 @@ def get_values_for_inverter(self):
 
         elif self.dtuvariant == constants.DTUVARIANT_TEMPLATE:
             # logging.debug("JSON data: %s" % meter_data)
-            power = float(get_nested(meter_data, self.custpower) * float(self.custpower_factor))
-            pvyield = float(get_nested(meter_data, self.custtotal) * float(self.custtotal_factor))
-            voltage = float(get_nested(meter_data, self.custvoltage))
-            dc_voltage = float(get_nested(meter_data, self.custdcvoltage))
-            current = float(get_nested(meter_data, self.custcurrent))
+            get_power = get_nested(meter_data, self.custpower)
+            power_value = try_get_value(get_power, float, self.custpower_default)
+            if isinstance(power_value, float) or isinstance(power_value, int):
+                power = float(power_value * float(self.custpower_factor))
```

Comment:
Another thought: Why do we do this replacment by defaults only for Templates? Should we find a place where to do it so it will work for inverters, too?

Reply:
That's ok. First step: Make it run and verify that is meets the requirement...